### PR TITLE
AR-60: Add Interactive atoms to CSP

### PR DESCRIPTION
**Pull request by dskamiotis**
## Why are you doing this?
iframed interactive atoms were being blocked by the CSP, this fixes the problem

## Screenshots

| Before | After |
| --- | --- |
|  |  |

### CODE REVIEW

File: src/components/atoms/interactiveAtom.tsx [hunk 1]
```diff
@@ -5,7 +5,8 @@ import { Format } from '@guardian/types/Format';
 import { getPillarStyles, PillarStyles } from 'pillarStyles';
 import { Option, map, withDefault } from 'types/option';
 import { pipe2 } from 'lib';
-
+import {pageFonts} from "styles";
+import {remSpace} from "@guardian/src-foundations";
```
Comment (webb04) [inline src/components/atoms/interactiveAtom.tsx:9]: For consistency?

```suggestion
import { pageFonts } from "styles";
import { remSpace } from "@guardian/src-foundations";
```

File: src/server/page.tsx [hunk 2]
```diff
@@ -26,6 +26,7 @@ import { Option, some, none, map, withDefault } from 'types/option';
 import { compose, pipe2 } from 'lib';
 import { csp } from 'server/csp';
 import { remSpace } from '@guardian/src-foundations';
+import {atomCss, atomScript} from "../components/atoms/interactiveAtom";
```
Comment (webb04) [inline src/server/page.tsx:29]: Does this work?

```suggestion
import { atomCss, atomScript } from "components/atoms/interactiveAtom";
```

File: src/components/atoms/interactiveAtom.tsx (hunk 1 above)
Comment (frankie297) [reply src/components/atoms/interactiveAtom.tsx:9]: This is intellij being it's weird and wonderful self

File: src/components/atoms/interactiveAtom.tsx [hunk 3]
```diff
@@ -23,6 +24,18 @@ const InteractiveAtomStyles = (pillarStyles: PillarStyles): SerializedStyles =>
         border-bottom: 0.0625rem solid ${neutral[86]};
     }
 `;
+const atomCss = `
+    ${pageFonts} 
+    body {
+        background: white !important;
+        padding: ${remSpace[2]} !important;
```
Comment (webb04) [inline src/components/atoms/interactiveAtom.tsx:31]: These styles should be wrapped in a `${darkModeCss``}`

Comment (webb04) [reply src/components/atoms/interactiveAtom.tsx:31]: Actually these styles are just a string and aren't wrapped in ``` css`` ``` so you would need:

`@media (prefers-color-scheme: dark) {`

File: src/renderer.ts [hunk 4]
```diff
@@ -5,7 +5,7 @@ import { css, jsx as styledH, SerializedStyles } from '@emotion/core';
 import { from, until } from '@guardian/src-foundations/mq';
 import { text as textColour, neutral } from '@guardian/src-foundations/palette';
 import { Option, fromNullable, some, none, andThen, map, withDefault } from 'types/option';
-import { basePx, icons, darkModeCss, pageFonts } from 'styles';
+import { basePx, icons, darkModeCss} from 'styles';
```
Comment (webb04) [inline src/renderer.ts:8]:
```suggestion
import { basePx, icons, darkModeCss } from 'styles';
```

File: src/renderer.ts [hunk 5]
```diff
@@ -20,7 +20,7 @@ import BodyImageThumbnail from 'components/bodyImageThumbnail';
 import FigCaption from 'components/figCaption';
 import BodyImageHalfWidth from 'components/bodyImageHalfWidth';
 import Anchor from 'components/anchor';
-import InteractiveAtom from 'components/atoms/interactiveAtom';
+import InteractiveAtom, {atomCss, atomScript} from 'components/atoms/interactiveAtom';
```
Comment (webb04) [inline src/renderer.ts:23]:
```suggestion
import InteractiveAtom, { atomCss, atomScript } from 'components/atoms/interactiveAtom';
```

**Pull request merged**